### DISCUSSION TITLE
How to right justify on decimal point

### DISCUSSION THREAD

**Discussion by candcconsulting** [iTwin/iTwinUI#790] [General]
The examples here 
https://itwin.github.io/iTwinUI-react/?path=/story/core-table--basic
are great but it is a common requirement to right justify on decimal places
If I add text-align as this discussion
https://stackoverflow.com/questions/53605342/how-to-make-some-columns-align-left-and-some-column-align-center-in-react-table
columns: [
           {             
              accessor: "firstName",
              Header: "S Column 1",
              style: {
                textAlign: 'right'
              }
           }
]
Then the text is not right aligned ... if it is added to the cell
<span style = {{textAlign: "right"}} title={props.value}>{props.value}</span>
Then the <span> does not fill the <div> of the cell
Is this possible and if so which style should be used ?

**Comment by veekeys** [comment]
Hey @candcconsulting,

We are using flex on our table, so textAlign will not work.
For horizontal alignment, you should use `justifyContent`. So in your case, try `justifyContent: 'flex-end'`.
Read more about flexbox model and specifically `justify-content` here
https://css-tricks.com/snippets/css/a-guide-to-flexbox/#aa-justify-content.

We could update our storybook to provide at least one story with numbers aligned to the right.
Feel free to create a PR with such update too!

**Reply by candcconsulting** [reply]
Does not seem to pass down 
![image](https://user-images.githubusercontent.com/19268085/198686757-fbcd1196-0ac0-4cf0-a3be-7c78a6fda4d5.png)
              accessor: "count",
                    Cell: SkeletonCell,
                    Header: "Count",
                    disableResizing: false,
                    Filter: tableFilters.NumberRangeFilter(),
                    style : {justifyContent: "flex-end"}

**Reply by mayank99** [reply]
you need to use the `Cell` property and apply the styles on the component you pass there

**Reply by candcconsulting** [reply]
Thanks ... I am not sure I did it the way @mayank99 proposed as I did not seem to have access to the cell component.  However, I did see in the samples that there was a DefaultCell 
![image](https://user-images.githubusercontent.com/19268085/198817703-9cbd80e1-c1d8-4df9-b65c-8f8a01b4b22a.png)
So I added this code to a cell renderer
export const numericCellRenderer = (
  props: CellRendererProps<any>,

) => {
  return (
    <DefaultCell
      {...props}
      style={{ justifyContent: "flex-end" }}
    />
  );
};
Which works OK ... I need to add some space
If you would like to show an example of how to do it through Cell that would be great, but an example of aligned on decimal point would be fantastic ...

**Reply by candcconsulting** [reply]
and this is ultimately what seemed to work best
```

export const numericCellRenderer = (
  props: CellRendererProps<any>,

) => {
  return (
    <DefaultCell
      {...props}
      style={{ justifyContent: "flex-end"}}
    />
  );
};

export type NumericalCellProps = PropsWithChildren<{
  value: any;
  row: {
    original: any;
  };
  style : any;
}>
export const NumericCell = (props: NumericalCellProps) => {
  
  if (Object.keys(props.row.original).length !== 0) {
    // props.style = {...props.style, ...{justifyContent: "flex-end"}}
    return (props.children ?? (
      <span  title={props.value}>{new Intl.NumberFormat('en-EN', { minimumFractionDigits: 3, maximumFractionDigits: 3 }).format(props.value)}&nbsp;</span>
    )) as React.ReactElement;
  }
  return <Body isSkeleton={true}>Fetching</Body>;
};
```
